### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
           command: src/test/resources/localstack/setup-es.sh
       - hmpps/wait_till_ready:
           port: 4566
-          host: es1.eu-west-2.es.localhost.localstack.cloud
+          host: es01.eu-west-2.es.localhost.localstack.cloud
```
Comment (steverendell) [inline .circleci/config.yml:34]: bizarrely the min no of chars is now 4 (i think thats a bug)

File: .gitignore [hunk 2]
```diff
@@ -33,3 +33,4 @@ dps-gradle-spring-boot-suppressions.xml
 # Helm
 helm_deploy/prisoner-offender-search/charts/
 Chart.lock
+/volume/
```
Comment (steverendell) [inline .gitignore:36]: this is where localstack now keeps its internal data

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt [hunk 3]
```diff
@@ -79,20 +82,33 @@ abstract class QueueIntegrationTest : IntegrationTest() {
   protected val hmppsEventsQueue by lazy { hmppsQueueService.findByQueueId("hmppseventtestqueue") ?: throw MissingQueueException("hmppseventtestqueue queue not found") }
 
   fun getNumberOfMessagesCurrentlyOnQueue(): Int? {
-    val queueAttributes = eventQueueSqsClient.getQueueAttributes(eventQueueUrl, listOf("ApproximateNumberOfMessages"))
-    return queueAttributes.attributes["ApproximateNumberOfMessages"]?.toInt()
+    val queueAttributes = eventQueueSqsClient.getQueueAttributes(eventQueueUrl, listOf("ApproximateNumberOfMessages", "ApproximateNumberOfMessagesNotVisible"))
+    val number = queueAttributes.attributes["ApproximateNumberOfMessages"]?.toInt()!! + queueAttributes.attributes["ApproximateNumberOfMessagesNotVisible"]?.toInt()!!
+    log.trace("Messages on event queue: {}", number)
+    return number
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt:88]: these now also include non-visible messages

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/HmppsDomainEventsEmitterIntTest.kt [hunk 4]
```diff
@@ -40,6 +40,8 @@ class HmppsDomainEventsEmitterIntTest : QueueIntegrationTest() {
     await untilCallTo { getNumberOfMessagesCurrentlyOnDomainQueue() } matches { it == 1 }
 
     val result = hmppsEventsQueue.sqsClient.receiveMessage(hmppsEventsQueue.queueUrl).messages.first()
+    hmppsEventsQueue.sqsClient.deleteMessage(hmppsEventsQueue.queueUrl, result.receiptHandle)
+
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/HmppsDomainEventsEmitterIntTest.kt:44]: needed to fully dequeue message

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResourceTest.kt [hunk 5]
```diff
@@ -35,7 +34,7 @@ class PrisonerIndexResourceTest : QueueIntegrationTest() {
     )
     resetStubs()
     setupIndexes()
-    Mockito.reset(indexQueueService)
+    // Mockito.reset(indexQueueService)
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResourceTest.kt:37]: whoops - will fully remove next time

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt [hunk 6]
```diff
@@ -163,9 +178,8 @@ abstract class QueueIntegrationTest : IntegrationTest() {
       .exchange()
       .expectStatus().isOk
 
-    await untilCallTo { prisonRequestCountFor("/api/offenders/${prisonerNumbers.last()}") } matches { it == 1 }
-
-    await untilCallTo { getNumberOfMessagesCurrentlyOnIndexQueue() } matches { it == 0 }
+    await.atMost(Duration.ofSeconds(60)) untilCallTo { prisonRequestCountFor("/api/offenders/${prisonerNumbers.last()}") } matches { it == 1 }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt:181]: Wow this implies the tests are running very slowly! I've never seen 10 seconds not be enough before.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResourceTest.kt (hunk 5 above)
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResourceTest.kt:37]: I'm surprised this isn't needed as I thought all SpyBeans were not reset automatically.

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/MessageIntegrationTest.kt [hunk 7]
```diff
@@ -93,7 +93,6 @@ class MessageIntegrationTest : QueueIntegrationTest() {
     eventQueueSqsClient.sendMessage(eventQueueUrl, message)
 
     await untilCallTo { getNumberOfMessagesCurrentlyOnQueue() } matches { it == 0 }
-    Thread.sleep(500)
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/MessageIntegrationTest.kt:96]: 🙌

**Pull request merged**